plugins/nextcloud/nextcloud_

```diff
@@ -0,0 +1,230 @@
+#!/bin/bash
+
+set -e
+
+: << =cut
+
+=head1 NAME
+
+nextcloud_ - Monitor usage of nextcloud instances
+
+=head1 APPLICABLE SYSTEMS
+
+Nexcloud instances
+
+=head1 CONFIGURATION
+
+Requires installed curl and jq, a command-line json processor.
+
+This is a wildcard plugin. To monitor a nextcloud instance, link
+nextcloud_<nextcloud-domain> to this file. You can even append a port 
+(:8443) to the file if needed. For example,
+
+  ln -s /usr/share/munin/plugins/nextcloud_ \
+        /etc/munin/plugins/nextcloud_cloud.domain.tld
+
+Set username and password in your munin-node configuration
+
+[nextcloud_cloud.domain.tld]
+env.username <nexcloud_user>
+env.password <nextcloud_password>
+
+It's advised to set an app password (for this plugin) in your nextcloud 
+instance and not to use the "real" password of your nextcloud user.
+
+=head1 AUTHOR
+
+Copyright (C) 2020 Sebastian L. (https://momou.ch)
+
+=head1 LICENSE
+
+GPLv2
+
+=head1 MAGIC MARKERS
+
+ #%# family=manual
+ #%# capabilities=autoconf
+
+=cut
+
+. "$MUNIN_LIBDIR/plugins/plugin.sh"
+
+API_PATH="/ocs/v2.php/apps/serverinfo/api/v1/info?format=json"
+DOMAIN="${0##*nextcloud_}"
+CLEANDOMAIN="$(clean_fieldname "$DOMAIN")"
+USERNAME="${username:-}"
+PASSWORD="${password:-}"
+
+print_json_data() {
+    [ -z "$1" ] && exit 0
```

Comment:
I don't know how to specify local variables in functions in posix shell, but I just used a simple (exposed) variable, which works.

Reply:
> I don't know how to specify local variables in functions in posix shell

I guess, `shellcheck` is confusing you here. Expecting `dash` as the shell implementation is reasonable (at least we do this in munin). Thus you should run `shellcheck -s dash` - this would void any warnings for the simple line `local KEY VALUE`.

> [..] but I just used a simple (exposed) variable, which works.

Yes it works, until it breaks by accident :)

Quoting the python people here (`python3 -m this | grep -i names`):
```
Namespaces are one honking great idea -- let's do more of those!
```

Anyway - since it works, it can be considered a matter of taste ...

Reply:
I like the "taste" ...